src/en/anchira/src/eu/kanade/tachiyomi/extension/en/anchira/Anchira.kt

```diff
@@ -0,0 +1,342 @@
+        var url = libraryUrl.toHttpUrl().newBuilder()
+
+        url.addQueryParameter("page", page.toString())
+
+        if (query.isNotBlank()) {
+            url.addQueryParameter("s", query)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is CategoryGroup -> {
+                    var sum = 0
+
+                    filter.state.forEach { category ->
+                        when (category.name) {
+                            "Manga" -> if (category.state) sum += 1
+                            "Doujinshi" -> if (category.state) sum += 2
+                            "Illustration" -> if (category.state) sum += 4
+                        }
+                    }
+
+                    if (sum > 0) url.addQueryParameter("cat", sum.toString())
+                }
+
+                is SortFilter -> {
+                    val sort = when (filter.state?.index) {
+                        0 -> "1"
+                        1 -> "2"
+                        2 -> "4"
+                        4 -> "32"
+                        else -> ""
+                    }
+
+                    if (sort.isNotEmpty()) url.addQueryParameter("sort", sort)
+                    if (filter.state?.ascending == true) url.addQueryParameter("order", "1")
+                }
+
+                is FavoritesFilter -> {
+                    if (filter.state) {
+                        if (!isLoggedIn()) {
+                            throw IOException("No login cookie found")
+                        }
+
+                        url = url.toString().replace("library", "user/favorites").toHttpUrl()
+                            .newBuilder()
+                    }
+                }
+
+                else -> {}
+            }
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response) = latestUpdatesParse(response)
+
+    // Details
+
+    override fun mangaDetailsRequest(manga: SManga) =
+        GET("$libraryUrl/${getPathFromUrl(manga.url)}", headers)
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val data = decodeBytes<Entry>(response.body)
+
+        return SManga.create().apply {
+            url = "/g/${data.id}/${data.key}"
+            title = data.title
+            thumbnail_url =
+                "$cdnUrl/${data.id}/${data.key}/b/${data.data[data.thumbnailIndex].name}"
+            artist = data.tags.filter { it.namespace == 1 }.joinToString(", ") { it.name }
+            author = data.tags.filter { it.namespace == 2 }.joinToString(", ") { it.name }
+            genre = prepareTags(data.tags)
+            update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+            status = SManga.COMPLETED
+        }
+    }
+
+    override fun getMangaUrl(manga: SManga) = if (preferences.openSource) {
+        val id = manga.url.split("/").reversed()[1].toInt()
```

Comment:
```suggestion
        val id = manga.url.split("/").last().toInt()
```

Use [`kotlin.collections.last`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/last.html), which is more idiomatic Kotlin.

Reply:
I need the 2nd to last here `/g/{ID}/{KEY}` which would be the ID.

Reply:
Ah, my bad, confused it for `.reversed()[0]`.